File: CPP/63.cpp
#include <iostream>
#include <math.h>
#include <limits>

using namespace std;

int main()
{
    int counter = 0;
    cout << numeric_limits<unsigned long long int>::max();
    cin.get();
    for (int i = 1; i < 50; i++)
    {
        int x = 1;
        while (1)
        {
            unsigned long long int number = pow(x, i);
            unsigned long long int number_2 = pow(x, i);
            int length = 0;
            while (number_2 > 0)
            {
                length++;
                number_2 /= 10;
            }
            if (length > i)
            {
                break;
            }
            else if (length == i)
            {
                cout << x << "^" << i << " = " << number << endl;
                counter++;
            }
            x += 1;
        }
    }
    cout << counter << endl;
	return 0;
}

File: Python/63.py
counter = 0
for i in range(100):
    x = 1
    while 1:
        number = x**i
        length = len(str(number))
        if length > i:
            break
        elif length == i:
            print str(number) + " = " + str(x) + "^" + str(i)
            counter += 1
        x += 1
print counter